website/addons/googledrive/tests/test_views.py

```diff
@@ -5,183 +5,80 @@
-        }
-        mock_auth_client_userinfo.return_value = {
-            'sub': 'unique id',
-            'name': 'test-user',
-        }
-        url = api_url_for('googledrive_oauth_finish', user_no_addon.auth, nid=self.project._primary_key, code='1234', state=state)
-        res = self.app.get(url)
-        assert_is_redirect(res)
-
-    def test_googledrive_oauth_finish_cancelled(self):
-        user_no_addon = AuthUserFactory()
-        url = api_url_for(
-            'googledrive_oauth_finish',
-            user_no_addon.auth,
-            nid=self.project._primary_key,
-            code='1234',
-            state='3322',
-            error='User declined!'
-        )
-        res = self.app.get(url)
-        assert_is_redirect(res)
-
-    @mock.patch('website.addons.googledrive.views.auth.GoogleAuthClient.userinfo')
-    @mock.patch('website.addons.googledrive.views.auth.GoogleAuthClient.finish')
-    @mock.patch('website.addons.googledrive.views.auth.session')
-    def test_googledrive_oauth_finish_user_only(self, mock_session, mock_auth_client_finish, mock_auth_client_userinfo):
-        user_no_addon = AuthUserFactory()
-        state = '1234'
-        mock_session.data = {
-            'googledrive_auth_state': state,
-        }
-        mock_auth_client_finish.return_value = {
-            'access_token': '1111',
-            'refresh_token': '2222',
-            'expires_at': time.time() + 3600,
-        }
-        mock_auth_client_userinfo.return_value = {
-            'sub': 'unique id',
-            'name': 'test-user',
-        }
-        url = api_url_for('googledrive_oauth_finish', user_no_addon.auth, code='1234', state=state)
-        res = self.app.get(url)
-        assert_is_redirect(res)
-
-    @mock.patch('website.addons.googledrive.views.auth.GoogleAuthClient.revoke')
-    def test_googledrive_oauth_delete_user(self, mock_auth_client_revoke):
-        self.user_settings.access_token = 'abc123'
-        self.user_settings.save()
-        assert_true(self.user_settings.has_auth)
-        self.user.save()
-        url = api_url_for('googledrive_oauth_delete_user')
-        self.app.delete(url)
-        self.user_settings.reload()
-        mock_auth_client_revoke.assert_called_once()
-        assert_false(self.user_settings.has_auth)
-
-    def test_googledrive_deauthorize(self):
-        self.node_settings.folder_id = 'foobar'
-        self.node_settings.folder_path = 'My folder'
-        self.node_settings.save()
-
-        url = self.project.api_url_for('googledrive_deauthorize')
-
-        self.app.delete(url)
-        self.project.reload()
-        self.node_settings.reload()
-
-        assert_false(self.node_settings.has_auth)
-        assert_is(self.node_settings.folder_id, None)
-        assert_is(self.node_settings.folder_path, None)
-        assert_is(self.node_settings.user_settings, None)
+from tests.factories import AuthUserFactory, ProjectFactory, ExternalAccountFactory
```

Comment:
`time`, `datetime`, and `tests.base.assert_is_redirect` are also unused and should be removed.